lib/BuildSystem/BuildSystem.cpp

```diff
@@ -1186,8 +1296,153 @@ class DirectoryTreeStructureSignatureTask : public Task {
   }
 
 public:
-  DirectoryTreeStructureSignatureTask(StringRef path, StringList&& filters)
-    : path(path), filters(std::move(filters)) {}
+  DirectoryTreeStructureSignatureTask(StringRef path) : path(path) {}
+};
+
+/// This is the task to "build" a directory node which will encapsulate (via a
+/// signature) a filtered view of the contents of the directory, recursively.
+class FilteredDirectoryTreeSignatureTask : public Task {
```

Comment:
I do wonder if we couldn't have kept a single task implementation, and just had the constructor take the filter list and then switch on which kind of nodes to request?

Reply:
Yeah, in the end the differences are not as great as I originally thought at the actual tree signature level.   It is probably worth combining the back together.